sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/V2ScanRelationPushDown.scala

```diff
@@ -255,7 +256,20 @@ object V2ScanRelationPushDown extends Rule[LogicalPlan] with PredicateHelper {
             sHolder.pushedLimit = Some(limitValue)
           }
           globalLimit
-        case _ => globalLimit
+        case _ =>
+          child transform {
+            case sort @ Sort(order, _, ScanOperation(_, filter, sHolder: ScanBuilderHolder))
+              if filter.length == 0 =>
+              val orders = DataSourceStrategy.translateSortOrders(order)
+              val topNPushed = PushDownUtils.pushTopN(sHolder.builder, orders.toArray, limitValue)
+              if (topNPushed) {
+                sHolder.pushedLimit = Some(limitValue)
+                sHolder.sortValues = orders
+              }
+              sort
```

Comment:
The limit with sort has been pushed done, Whether the sort node can be remove?

Reply:
Some database supports partition sort.

Reply:
Yea I think we can remove the sort node from the query plan.